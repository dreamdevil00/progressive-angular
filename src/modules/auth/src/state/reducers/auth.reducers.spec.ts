import {
  reducer,
  AuthState,
  Credential,
  User,
  initialAuthState,
  AuthLoginAction,
  AuthLoginSuccessAction,
  AuthGetRolesSuccessAction,
} from '../../index';
import 'jest';

describe('AuthReducer', () => {

  describe('undefined action', () => {
    it('should return the default state', () => {
      const result = reducer(void 0, {} as any);
      expect(result).toMatchSnapshot();
    });
  });

  describe('wrong login payload', () => {
    it('sould Not authenticate a user', () => {
      const user: Credential = { username: 'username', password: 'password' };
      const expectedResult = initialAuthState;
      const result = reducer(initialAuthState, new AuthLoginAction(user));
      expect(result).toMatchSnapshot();
    });
  });

  describe('LOGIN_SUCCESS', () => {
    it('should set login to true, user to user info', () => {
      const user = { id: 'testId', token: 'testToken'};
      const expectResult = {
        id: 'testId',
        token: 'testToken',
        roles: [],
        login: true,
      };
      const result = reducer(initialAuthState, new AuthLoginSuccessAction(user));
      expect(result).toMatchSnapshot();
    });
  });

  describe('GET_ROLES_SUCCESS', () => {
    it('should  set user roles', () => {
      const user = { roles: ['admin']};
      const expectResult = {
        id: null,
        token: null,
        roles: ['admin'],
        login: false,
      };
      const result = reducer(initialAuthState, new AuthGetRolesSuccessAction(user));
      expect(result).toMatchSnapshot();
    });
  });
});
