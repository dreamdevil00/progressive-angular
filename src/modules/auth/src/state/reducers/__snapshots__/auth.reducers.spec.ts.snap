// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthReducer GET_ROLES_SUCCESS should  set user roles 1`] = `
Object {
  "id": null,
  "login": false,
  "roles": Array [
    "admin",
  ],
  "token": null,
}
`;

exports[`AuthReducer LOGIN_SUCCESS should set login to true, user to user info 1`] = `
Object {
  "id": "testId",
  "login": true,
  "roles": Array [],
  "token": "testToken",
}
`;

exports[`AuthReducer undefined action should return the default state 1`] = `
Object {
  "id": null,
  "login": false,
  "roles": Array [],
  "token": null,
}
`;

exports[`AuthReducer wrong login payload sould Not authenticate a user 1`] = `
Object {
  "id": null,
  "login": false,
  "roles": Array [],
  "token": null,
}
`;
